src/App.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import './App.css';
+import { GoodsList } from './components';
+
+const goodsFromServer = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+class App extends React.Component {
+  state = {
+    goods: [],
+    isReversed: false,
+    sortBy: '',
+  };
+
+  start = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+    });
+  };
+
+  sortByName = () => {
+    this.setState({ sortBy: 'name' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  render() {
+    const { goods, sortBy, isReversed } = this.state;
+    const visibleGoods = [...goods];
```

Comment:
```suggestion

```
You did a copy of the goods array twice.  (27 row in first)

Reply:
yeah, right